lib/builder/builder.js

```diff
@@ -62,20 +62,31 @@ export default class Builder {
     // }
   }
 
+  pluginBasename (src) {
+    return path.basename(src, path.extname(src))
+      .replace(/[^a-zA-Z?\d\s:]/g, '')
+  }
+
   get plugins() {
+    let exists
     return _.uniqBy(
       this.options.plugins.map((p) => {
-        if (typeof p === 'string') p = { src: p }
-        const pluginBaseName = path.basename(p.src, path.extname(p.src)).replace(
-          /[^a-zA-Z?\d\s:]/g,
-          ''
-        )
-        return {
-          src: this.nuxt.resolveAlias(p.src),
-          ssr: p.ssr !== false,
-          name: 'nuxt_plugin_' + pluginBaseName + '_' + hash(p.src)
+        if (typeof p === 'string') {
+          p = { src: p }
+        }
+        p.basename = this.pluginBasename(p.src)
+        return p
+      }).filter((p) => {
+        exists = fs.existsSync(this.nuxt.resolveAlias(p.src))
+        if (!exists) {
+          consola.warn(`Plugin not found: ${p.basename}`)
         }
-      }),
+        return exists
+      }).map((p) => ({
+        src: this.nuxt.resolveAlias(p.src),
+        ssr: p.ssr !== false,
+        name: 'nuxt_plugin_' + p.basename + '_' + hash(p.src)
```

Comment:
Any reason for moving base name calculation above?

Reply:
@clarkdo so we don't do it twice. I need that same calculation to display any potential warnings in the preceding `filter()`.